motioneye/mjpgclient.py

```diff
@@ -241,15 +240,13 @@ def _on_www_authenticate(self, future: Future) -> None:
             logging.debug('mjpg client using basic authentication')
 
             auth_header = utils.build_basic_header(self._username, self._password)
-            w_data = (
-                b'GET / HTTP/1.0\r\nAuthorization: %s\r\nConnection: close\r\n\r\n'
-                % auth_header
-            )
+            w_data = f'GET / HTTP/1.0\r\nAuthorization: {auth_header}\r\nConnection: close\r\n\r\n'.encode()
             w_future = utils.cast_future(self.write(w_data))
             w_future.add_done_callback(self._seek_http)
 
             return
 
+        data = data.decode()
         if data.startswith('Digest'):
```

Comment:
Does this work on bytes? Else probably `b'Digest'`?

Reply:
data is decoded above so it's string. If you mean that?

Inside if there is nested splits, need to check if all of those fail without b''.

Reply:
Ah sorry, too early in the morning, in my head it was `.encode()` 😄.